randovania/game_connection/connector/dread_remote_connector.py

```diff
@@ -66,6 +107,21 @@ def new_player_location_received(self, state_or_region: str):
                                         None)
         self.PlayerLocationChanged.emit(PlayerLocationEvent(self.current_region, None))
 
+    def new_collected_locations_received(self, new_indices: bytes):
+        locations = set()
+        if new_indices.startswith(b"locations:"):
+            index = 0
+            for c in new_indices[len(b"locations:"):]:
+                for i in range(8):
+                    if c & (1 << i):
+                        locations.add(PickupIndex(index))
+                    index += 1
+        else:
+            self.logger.warning(f"Unknown response: {new_indices}")
```

Comment:
dark has taught mii that it's best practice to avoid fstrings in logging, since it'll always evaluate the fstring even if the log is ignored at the current log level. another form of string formatting would be more appropriate for this and other logging calls throughout the pr

Reply:
Hmm what would be the proper way for doing that? The code is send in chunks because of the buffer limit of 4096. It get blocks of `all_code` and concats them as long as they not exceed the limit.
If I just move this to one lua file, I have to make sure that it is still in some sort of blocks.
And also the dynamcially created parts from the region_list or resource_database of the game needs to be filled.